src/analyses/interval_domain.h

```diff
@@ -40,10 +40,15 @@ class interval_domaint:public ai_domain_baset
     const ai_baset &ai,
     const namespacet &ns) const final;
 
+  bool join(const interval_domaint &b);
```

Comment:
Join is the mathematical operation on the domains.  We use this (in merge) to merge together different states.  There are uses of join that are not merging (as in ai_simplify) and ways of merging (like delayed joins) that are not (just) join.

... and no-one who hasn't read that or a lot of AI literature will understand that so I'll add it to the comments :-)